pkg/install/install.go

```diff
@@ -244,6 +253,57 @@ func (o *Install) Run(ctx context.Context) error {
 	return nil
 }
 
+func (o *Install) supportedVersion(meta *versionpb.MetadataVersion) (*supportedVersion, error) {
+	supVer := &supportedVersion{}
+
+	// Parse MetadataVersion into supportedVersion struct.
+	config := map[string]*goversion.Constraints{
+		"kubernetes": &supVer.kubernetes,
+	}
+	for key, ref := range config {
+		if s, ok := meta.GetSupported()[key]; ok {
+			c, err := goversion.NewConstraint(s)
+			if err != nil {
+				return nil, errors.Join(err, fmt.Errorf("invalid %s constraint %s", key, s))
+			}
+			*ref = c
+		}
+	}
+
+	return supVer, nil
+}
+
+func (o *Install) checkRequirements(meta *versionpb.MetadataVersion) error {
+	supVer, err := o.supportedVersion(meta)
+	if err != nil {
+		return err
+	}
+
+	if len(supVer.kubernetes) > 0 {
+		o.l.Info("Checking Kubernetes version requirements")
+		k8sVersionInfo, err := o.kubeClient.GetServerVersion()
+		if err != nil {
+			return errors.Join(err, errors.New("could not retrieve Kubernetes version"))
+		}
+
+		k8sVersion, err := goversion.NewVersion(fmt.Sprintf("%s.%s", k8sVersionInfo.Major, k8sVersionInfo.Minor))
+		if err != nil {
+			return errors.Join(err, errors.New("invalid Kubernetes version"))
+		}
+
+		if !supVer.kubernetes.Check(k8sVersion) {
+			return fmt.Errorf(
+				"kubernetes version %q does not meet minimum requirements of %q",
+				k8sVersion.String(), supVer.kubernetes.String(),
+			)
+		}
+
+		o.l.Debugf("Finished requirements check for Kubernetes version")
+	}
```

Comment:
Can we use the `checkK8sRequirements()` function defined in the upgrade module, or vice-versa?

Reply:
Just trying to avoid duplicate code :slightly_smiling_face:

Reply:
we could refactor, but it required extracting upgrade/install structures into a common package so I decided it's maybe better not to touch this behavior so we don't break it and make the PR smaller.

Reply:
It's not a big deal but I think it's still work considering this refactor but it can be done in a follow-up PR.